packages/fuel-indexer-macros/src/indexer.rs

```diff
@@ -442,15 +454,19 @@ fn process_fn_items(
             fn selector_to_type_id(&self, sel: u64) -> usize {
                 match sel {
                     #(#abi_selectors)*
-                    // TODO: should handle this a little more gently
-                    _ => panic!("Unknown type id."),
+                    _ => {
+                        Logger::warn("Unknown selector; check ABI to make sure function outputs match to types");
+                        0
+                    }
                 }
             }
 
             fn decode_type(&mut self, ty_id: usize, data: Vec<u8>) {
                 match ty_id {
                     #(#abi_decoders),*
-                    _ => panic!("Unkown type id '{}'.", ty_id),
+                    _ => {
+                        Logger::warn("Unknown type ID; check ABI to make sure types are well-formed");
```

Comment:
- Same here. If we're trying to decode unknown type IDs that's a sign of a bigger issue. 
- Still want to panic